org-code-javabuilder/playground/src/main/java/org/code/playground/TextItem.java

```diff
@@ -164,13 +160,25 @@ public double getRotation() {
   @Override
   protected HashMap<String, String> getDetails() {
     HashMap<String, String> details = super.getDetails();
-    details.put(TEXT_KEY, this.getText());
-    details.put(COLOR_RED_KEY, Integer.toString(this.colorRed));
-    details.put(COLOR_GREEN_KEY, Integer.toString(this.colorGreen));
-    details.put(COLOR_BLUE_KEY, Integer.toString(this.colorBlue));
-    details.put(FONT_KEY, this.getFont().toString());
-    details.put(FONT_STYLE_KEY, this.getFontStyle().toString());
-    details.put(ROTATION_KEY, Double.toString(this.getRotation()));
+    details.put(ClientMessageDetailKeys.TEXT, this.getText());
+    this.addColorToDetails(details);
+    details.put(ClientMessageDetailKeys.FONT, this.getFont().toString());
+    details.put(ClientMessageDetailKeys.FONT_STYLE, this.getFontStyle().toString());
+    details.put(ClientMessageDetailKeys.ROTATION, Double.toString(this.getRotation()));
     return details;
   }
+
+  private void setColorHelper(Color color) {
```

Comment:
Not exactly related to this change but since we split out color into its components here, should we do the same for `getColor` too? i.e. return a new color with the RGB values instead of original color object?

Reply:
I'll ask curriculum. My thought was it would be confusing for students to have a different object returned by get, but since we lock in the rgb values it might be confusing to have the same object too

Reply:
so it turns out we'll need to do some reworking of the color messages which I think would be best as a follow-up PR. [slack conversation](https://codedotorg.slack.com/archives/C01EF4GJ9GE/p1632255167336300) I'll leave this as-is for now

Reply:
Late to the party, but was just reading that slack thread...we want `textItem.getColor().setRed()` to set an item's color? Does that mean that `getColor()` would return the `textItem`?

Reply:
Oh nvm, it's returning the text item's color object, which we're setting to another color

Reply:
Wrapping my head around how we have colors set up and made a couple silly comments I just deleted, ignore them :).

Reply:
sorry for the confusion that was not exactly the correct syntax in the thread. It would be something like `textItem.getColor().setRed(255)`. The Color object has red, green and blue values. So we would be setting the red value to some new value, and want the text color to change

Reply:
Got it, should have looked at the Color class. Thanks!